adapters/http-base/src/main/java/org/eclipse/hono/adapter/http/JwtAuthHandler.java

```diff
@@ -0,0 +1,166 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.adapter.http;
+
+import java.net.HttpURLConnection;
+import java.util.Objects;
+
+import org.eclipse.hono.adapter.HttpContext;
+import org.eclipse.hono.adapter.auth.device.DeviceCredentialsAuthProvider;
+import org.eclipse.hono.adapter.auth.device.ExecutionContextAuthHandler;
+import org.eclipse.hono.adapter.auth.device.PreCredentialsValidationHandler;
+import org.eclipse.hono.adapter.auth.device.jwt.CredentialsParser;
+import org.eclipse.hono.adapter.auth.device.jwt.DefaultJwsValidator;
+import org.eclipse.hono.adapter.auth.device.jwt.JwtCredentials;
+import org.eclipse.hono.client.ClientErrorException;
+import org.eclipse.hono.client.ServiceInvocationException;
+import org.eclipse.hono.util.CredentialsConstants;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.MalformedJwtException;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.User;
+import io.vertx.ext.auth.authentication.AuthenticationProvider;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.HttpException;
+import io.vertx.ext.web.handler.impl.HTTPAuthorizationHandler;
+
+/**
+ * An auth handler for extracting an {@value CredentialsConstants#FIELD_AUTH_ID},
+ * {@value CredentialsConstants#FIELD_PAYLOAD_TENANT_ID} and JSON Web Token (JWT) from an HTTP context.
+ */
+public class JwtAuthHandler extends HTTPAuthorizationHandler<AuthenticationProvider> implements CredentialsParser {
+
+    private final PreCredentialsValidationHandler<HttpContext> preCredentialsValidationHandler;
+
+    /**
+     * Creates a new handler for an authentication provider and a Tenant service client.
+     *
+     * @param authProvider The authentication provider to use for verifying the device identity.
+     * @param realm The realm name.
+     * @throws NullPointerException if client auth is {@code null}.
```

Comment:
what is *client auth*?

Reply:
Good catch, this wasn't supposed to be still in there. I removed it.